### DISCUSSION TITLE
difference between handling session in handle.js and layout.server.js

### DISCUSSION THREAD

**Discussion by weepy** [sveltejs/kit#6688] [General]
I'm a bit confused here as there's sorta two ways of doing this - and I'm not quite sure on the particular features of each.

Is there any particular functional difference between using handle: 

```js
// hooks.server.js
export async function handle({ event, cookies }) {  
  const response = await resolve(event)
  events.locals.session  = await getSession( cookies.get("sid"))
  return response
}

//  +layout.server.js
export async function load({ locals }) {    
    const session = locals.session
   
    return { session } 
}
```

vs just doing it all in the layout

```js
// +layout.server.js
export async function load({ cookies }) {    
    const session = await getSession( cookies.get("sid"))

    return { session } 
}
```

**Comment by weepy** [comment]
I suppose in the former the locals would be available in the actions - where are in the layout they would only be available in the rendering. ?

**Comment by teemingc** [comment]
Since `hooks.server.js` runs for every request, it would call `getSession( cookies.get("sid"))` for every request (if that's what you want).

Whereas, doing it all in `+layout.server.js` will only run once or upon certain conditions that would trigger the `load` function to re-run. You also get the benefit of scoping with layout resets and so on.

> ```js
> // hooks.server.js
> export async function handle({ event, cookies }) {  
>   const response = await resolve(event)
>   events.locals.session  = await getSession( cookies.get("sid"))
>   return response
> }

If I'm not mistaken, you need to add to `event.locals` before passing it in to `resolve(event)`.

**Reply by weepy** [reply]
ok great thanks! - so it seems more convenient to run it directly in layout. ^_^
